### PR TITLE
[data ingestion] blob upload worker

### PR DESCRIPTION
**Pull request by phoenix-o**
replace current S3 upload worker with more general one based on object_store library

### CODE REVIEW

File: crates/sui-data-ingestion/src/executor.rs [hunk 1]
```diff
@@ -22,15 +22,15 @@ pub struct IndexerExecutor<P> {
...
     progress_store: ProgressStoreWrapper<P>,
-    pool_progress_sender: mpsc::Sender<(String, CheckpointSequenceNumber)>,
-    pool_progress_receiver: mpsc::Receiver<(String, CheckpointSequenceNumber)>,
+    pool_progress_sender: mpsc::UnboundedSender<(String, CheckpointSequenceNumber)>,
+    pool_progress_receiver: mpsc::UnboundedReceiver<(String, CheckpointSequenceNumber)>,
     metrics: DataIngestionMetrics,
 }
 
 impl<P: ProgressStore> IndexerExecutor<P> {
+    #[allow(clippy::disallowed_methods)] // channel is logically bounded by number_of_worker_jobs * MAX_CHECKPOINTS_IN_PROGRESS
```
Comment (bmwill) [inline crates/sui-data-ingestion/src/executor.rs:31]: do you have more context on why this change is necessary?

Comment (phoenix-o) [reply crates/sui-data-ingestion/src/executor.rs:31]: Currently, all channels have a capacity of `MAX_CHECKPOINTS_IN_PROGRESS`, and we ensure on the reader side that we won't produce more tasks than that. 
However, there's only one channel between all worker pools and the executor. 
While it's not necessary that every progress update is reported separately, in theory, it's possible, and then that channel can get congested, filled with up to `number_of_worker_jobs * MAX_CHECKPOINTS_IN_PROGRESS` messages.
The channel is still logically bounded, but because the framework is dynamic, and we register workflows as we go, we might not know the capacity in advance. Once the channel is created, we can't resize it. 
Alternatively, we can pass the limit of job types to the `new` method of the executor. That should work too, but it's more verbose from an API POV.

Comment (phoenix-o) [reply crates/sui-data-ingestion/src/executor.rs:31]: this is slightly orthogonal to the previous conversation. It's just to prevent a possible deadlock. Here we have multi producer single consumer situation, the only issue we're fixing is unknown(at least in advance) capacity

Comment (sadhansood) [inline crates/sui-data-ingestion/src/executor.rs:31]: Should we bound the channel to size `MAX_NUM_WORKER_JOBS * MAX_CHECKPOINTS_IN_PROGRESS`? Basically, suggesting to keep some theoretical upper bound (to prevent unbounded mem growth) vs an unbounded channel.

Comment (phoenix-o) [reply crates/sui-data-ingestion/src/executor.rs:31]: yes, but we don't know the value in advance with the existing API.
But we can change the API

**Review by sadhansood — APPROVED** [review]
